core/src/main/scala/kafka/tools/GetOffsetShell.scala

```diff
@@ -224,9 +227,14 @@ object GetOffsetShell {
   /**
    * Return the partition infos. Filter them with topicPartitionFilter.
    */
-  private def listPartitionInfos(consumer: KafkaConsumer[_, _], topicPartitionFilter: PartitionInfo => Boolean): Seq[PartitionInfo] = {
-    consumer.listTopics.asScala.values.flatMap { partitions =>
-      partitions.asScala.filter(topicPartitionFilter)
+  private def listPartitionInfos(client: Admin, topicPartitionFilter: PartitionInfo => Boolean, listInternal: Boolean): Seq[PartitionInfo] = {
+    val topics = client.listTopics(new ListTopicsOptions().listInternal(listInternal)).names().get().asScala
```

Comment:
I agree with your suggestion. It would be great if we could filter out the list of topics before describing them. Is it complicated?

Reply:
I changed the method from creating a topic-partition filter to create a topic filter and a topic-partition filter, the topic filter is used to filter topics before fetching and the topic-partition filter is used to filter topic-partitions after receiving topic metadata. PTAL.